src/borg/helpers.py

```diff
@@ -916,7 +916,12 @@ def __str__(self):
     def to_key_filename(self):
         name = re.sub('[^\w]', '_', self.path).strip('_')
         if self.proto != 'file':
-            name = self.host + '__' + name
+            name = re.sub('[^\w]', '_', self.host) + '__' + name
+        if len(name) > 100:
+            # Limit file names to some reasonable length. Most file systems
+            # limit them to 255 [unit of choice]; due to variations in unicode
+            # handling we truncate to 100 *characters*.
+            name = name[:100]
```

Comment:
Well, if done like that, it will truncate the path part and if you're unlucky, 2 different repos will map to same key filename (if they reside on same server, same base path, but different final path elements).